### ISSUE TITLE
Passing a Supply :$err to TAP::Harness results in  "No such method 'done' for invocant of type 'Supply'"

### ISSUE THREAD

**Issue by ugexe** [Raku/tap-harness6#47]
The `Supply` handling of `:$err` appears to be broken since there is no `.done` method on a supply as used here:
https://github.com/Raku/tap-harness6/blob/d0f95e080189ecc68655b9bbd7b0c8d25340ab91/lib/TAP.pm#L828-L830

```
# ran from root of zef repository

raku -e 'use TAP; my $supplier = Supplier.new; my $supply = $supplier.Supply; my @handlers = TAP::Harness::SourceHandler::Raku.new(:incdirs(["$*CWD/lib"])); my $parser = TAP::Harness.new(:@handlers, :err($supply)); my $p = $parser.run("t/00-load.t"); say $p.result'

t/00-load.t ..2/2Unhandled exception in code scheduled on thread 6
No such method 'done' for invocant of type 'Supply'.  Did you mean any
of these: 'none', 'one', 'clone', 'do', 'does', 'note'?
  in block  at /Users/ugexe/.rakubrew/versions/moar-2021.12/install/share/perl6/site/sources/6FEC28FA0762AC132736C81C67CCDB823E5E83FA (TAP) line 829
```

**Comment by Leont** [comment]
Yeah, that should be using a `Supplier` instead. I think this part of the code predates that split.

**Comment by Leont** [comment]
Should be fixed in 0.2.2
